src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/Harness/Listeners.cs

```diff
@@ -0,0 +1,433 @@
+    /// and the payload values, and we subclass this for the ETW case and for the EventListener case. 
+    /// </summary>
+    public abstract class Event
+    {
+        public virtual bool IsEtw { get { return false; } }
+        public virtual bool IsEventListener { get { return false; } }
+        public abstract string ProviderName { get; }
+        public abstract string EventName { get; }
+        public abstract object PayloadValue(int propertyIndex, string propertyName);
+        public abstract int PayloadCount { get; }
+        public virtual string PayloadString(int propertyIndex, string propertyName)
+        {
+            return PayloadValue(propertyIndex, propertyName).ToString();
+        }
+        public abstract IEnumerable<string> PayloadNames { get; }
+
+#if DEBUG
+        /// <summary>
+        /// This is a convenience function for the debugger.   It is not used typically 
+        /// </summary>
+        public List<object> PayloadValues 
+        {  
+            get
+            {
+                var ret = new List<object>();
+                for (int i = 0; i < PayloadCount; i++)
+                    ret.Add(PayloadValue(i, null));
+                return ret;
+            }
+        }
+#endif
+
+        public override string ToString()
+        {
+            StringBuilder sb = new StringBuilder();
+            sb.Append(ProviderName).Append('/').Append(EventName).Append('(');
+            for (int i = 0; i < PayloadCount; i++)
+            {
+                if (i != 0)
+                    sb.Append(',');
+                sb.Append(PayloadString(i, null));
+            }
+            sb.Append(')');
+            return sb.ToString();
+        }
+    }
+
+#if USE_ETW
+    /**************************************************************************/
+    /* Concrete implementation of the Listener abstraction */
+
+    /// <summary>
+    /// Implementation of the Listener abstraction for ETW.  
+    /// </summary>
+    public class EtwListener : Listener
+    {
+        internal static void InsureStopped()
+        {
+            using (var session = new TraceEventSession("EventSourceTestSession", "EventSourceTestData.etl"))
+                session.Stop();
+        }
+
+        public EtwListener(string dataFileName = "EventSourceTestData.etl", string sessionName = "EventSourceTestSession")
+        {
+            m_dataFileName = dataFileName;
+
+            // Today you have to be Admin to turn on ETW events (anyone can write ETW events).   
+            if (TraceEventSession.IsElevated() != true)
+            {
+                Assert.Inconclusive("Need to be admin elevated to run this test.");
+                throw new ApplicationException("Need to be elevated to run. ");
+            }
+
+            if (dataFileName == null)
+            {
+                Console.WriteLine("Creating a real time session " + sessionName);
+
+                Task.Factory.StartNew(delegate()
```

Comment:
Any case where you have ```Task.Factory.StartNew``` without arguments besides the delegate should be replaced by ```Task.Run```.  The key reason for this is xunit's TaskScheduler, which limits concurrency, and StartNew by default targets TaskScheduler.Current, whereas Task.Run targets TaskScheduler.Default.